### ISSUE TITLE
Can the compiler be packaged as an IDE plug-in?

### ISSUE THREAD

**Issue by Folds** [Folds/english#12]
The compiler can be run from the command line.

Currently, the compiler is part of the desktop application folder, and relies on features in the noodle library folder.  Both of these folders contain many definitions that are not needed by a pure compiler or IDE plug-in.

**Comment by elenderg** [comment]
Yes it can. I've already done it myself, you are interested, send me a message.

**Comment by elissongois** [comment]
You can create an IDE extension/plugin that executes the compile command through your IDE's built-in terminal using the working folder as a parameter. There is a fork that already does this. It includes a customized version for Notepad++ and a syntax highlighting extension for VS Code. The Portuguese version presents many error messages that encompass many cases of compilation errors and displays the contents of the line of code where the compilation process stopped along with the file name and line number.
